src/wallet/rpcdump.cpp

```diff
@@ -74,19 +77,166 @@ std::string DecodeDumpString(const std::string &str) {
     return ret.str();
 }
 
+bool SweepPrivKey(CBitcoinSecret key, CBitcoinAddress destAddress) {
+    // Get the private key to be swept and validate it
+    CKey dirtyKey = key.GetKey();
+    if (!dirtyKey.IsValid())
+        return false;
+
+    // Generate public key and verify it
+    CPubKey pubkey = dirtyKey.GetPubKey();
+    if (!dirtyKey.VerifyPubKey(pubkey))
+        return false;
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    // Get outputs from wallet
+    std::vector<COutput> vCoins;
+    pwalletMain->AvailableCoins(vCoins, false);
+
+    // Find all outputs of the key
+    std::vector<COutput> vToSweep;
+    BOOST_FOREACH(COutput output, vCoins)
```

Comment:
We are moving towards using the cpp11 for loop, no need to go back in new code and use a framework.